src/org/etools/j1939_84/controllers/part11/Part11Step13Controller.java

```diff
@@ -179,51 +177,45 @@ private void testSp12783(OBDModuleInformation module) {
-                if (getEngineModelYear() >= 2022 && getEngineModelYear() <= 2023) {
-                    addWarning("6.11.13.20.b - No response was received from "
-                            + module.getModuleName() + " for PG "
-                            + pg);
-                }
-            } else {
-                var partTwoPacket = get(packetForPg.getPgnDefinition().getId(), module.getSourceAddress(), 2);
-                hybridPacketForPg.getSpns()
-                                 .forEach(spn -> {
-                                     // 6.11.13.20.c - Fail each PG query where any active technology label or
-                                     // accumulator value received is greater than FAFFh, respectively.
-                                     if (spn.getRawValue() > 0xFAFFL) {
-                                         addFailure("6.11.13.20.c - Bin value received is greater than 0xFAFF(h) from "
-                                                 + module.getModuleName() + " for "
-                                                 + spn);
-                                     }
-                                     // 6.11.13.20.d -  Fail all values where the corresponding value received in part 2
-                                     // is greater than the part 11 value. (Where supported)
-                                     var partTwoValue = partTwoPacket.getSpnValue(spn.getId()).orElse(NOT_AVAILABLE);
-                                     if (partTwoValue > spn.getValue()) {
-                                         addFailure("6.11.13.20.d - Value received from " + module.getModuleName()
-                                                            + " for " + spn
-                                                            + " in part 2 was greater than part 11 value");
-                                     }
-
-                                 });
+        if (hybridChargeOpsPackets.isEmpty()) {
+            // 6.11.13.20.a - For MY2024+ Plug-in HEV DRIVES, Fail each PG query where
+            // no response was received.
+            if (getEngineModelYear() >= 2024) {
+                addFailure("6.11.13.20.a - No response was received from "
+                        + module.getModuleName());
+            }
+            // 6.11.13.20.b - For MY2022-23 Plug-in HEV DRIVES, Warn each PG query,
+            // where no response was received
+            if (getEngineModelYear() >= 2022 && getEngineModelYear() <= 2023) {
+                addWarning("6.11.13.20.b - No response was received from "
+                        + module.getModuleName());
             }
+        } else {
+            hybridChargeOpsPackets.forEach(genericPacket -> {
+                var partTwoPacket = get(genericPacket.getPgnDefinition().getId(), module.getSourceAddress(), 2);
+                genericPacket.getSpns().forEach(spn -> {
+                    // 6.11.13.20.c - Fail each PG query where any active technology label or
```

Comment:
I will to get Eric to update the document but for now, I did a check of SPN size because the verbiage changed to "6.11.13.20.c - Fail each PG query where any active technology label or accumulator value received is greater than FAFFh, respectively."